src/QMCWaveFunctions/lcao/MultiQuinticSpline1D.h

```diff
@@ -123,60 +136,73 @@ struct MultiQuinticSpline1D
     {
       const value_type dr=r-myGrid.lower_bound;
       const value_type* restrict a=(*Coeffs)[0];
-      for(size_t i=0; i<num_splines; ++i)
+      for(size_t i=0; i<num_splines_; ++i)
         u[i]=a[i]+first_deriv[i]*dr;
     }
-#if 0
     else if(r>=r_max)
     {
-      constexpr T czero(0);
-      for(size_t i=0; i<num_splines; ++i)
-        u[i]=czero;
+      // std::cout << "MultiQuinticSpline1D r>=r_max " << r << " >= "
+      // 		<< r_max << '\n';
+      assignLimitValues(u);
     }
-#endif
     else
     {
       int loc;
       const auto cL=myGrid.getCLForQuintic(r,loc);
       const size_t offset=loc*6;
+      //Coeffs is an OhmmsMatrix and [] is a row access operator
+      //returning a pointer to 'row' which is normal type pointer []
       const value_type* restrict a=(*Coeffs)[offset+0];
       const value_type* restrict b=(*Coeffs)[offset+1];
       const value_type* restrict c=(*Coeffs)[offset+2];
       const value_type* restrict d=(*Coeffs)[offset+3];
       const value_type* restrict e=(*Coeffs)[offset+4];
       const value_type* restrict f=(*Coeffs)[offset+5];
-      for(size_t i=0; i<num_splines; ++i)
+      for(size_t i=0; i<num_splines_; ++i)
         u[i]=a[i]+cL*(b[i]+cL*(c[i]+cL*(d[i]+cL*(e[i]+cL*f[i]))));
     }
   }
 
-  inline void evaluate(T r, T* restrict u, T* restrict du, T* restrict d2u) 
+  /** Approximation of getting the asymtotic constant spline values
+   */ 
+  inline void assignLimitValues(T* u)
+  {
+    int loc = myGrid.size() - 1;
+    const size_t offset=loc*6;
+    const value_type* restrict a=(*Coeffs)[offset+0];
+    for(size_t i=0; i<num_splines_; ++i)
+      u[i]=a[i];
```

Comment:
This doesn't seem right. All the AO are expected to be zero beyond r_max.

Reply:
this now zeros